### DISCUSSION TITLE
Docker SWARM secrets

### DISCUSSION THREAD

**Discussion by Rui-M-S-Ferreira** [DependencyTrack/dependency-track#1609] [General]
Dear all,

I´m trying to deploy Dependecy Track with external PSQL database.
For security reasons I don't want to specify the user and passwor don the environment variables, so I try to use secrets.

However the container cant read the secrets from /run/secrets/my_external_secret

Hi have this  on docker-compose.yml

version: '3.8'
...

secrets:
  DTRACK_DATABASE_PASSWORD:
    external: true
  DTRACK_DATABASE_USERNAME:
    external: true

services:

  dtrack-apiserver:
  ...
  secrets:
      - DTRACK_DATABASE_PASSWORD
      - DTRACK_DATABASE_USERNAME
  environment:
    #### ENABLING EXTERNAL DATABASE ####
      - ALPINE_DATABASE_MODE=external
      - ALPINE_DATABASE_URL=jdbc:postgresql://psql:5432/dtrack
      - ALPINE_DATABASE_DRIVER=org.postgresql.Driver
      - ALPINE_DATABASE_USERNAME=/run/secrets/DTRACK_DATABASE_USERNAME
      - ALPINE_DATABASE_PASSWORD=/run/secrets/DTRACK_DATABASE_PASSWORD

**Comment by jgstechnology** [comment]
If you are doing this within Azure you could use a Key vault backed variable group to inject the variables into a pipeline deployment.
